File: javascript/shuffle-string.js
function restoreString(s, indices) {
  let ans = new Array(s.length).fill(0);

  for (let i = 0; i < s.length; i++) {
    ans[indices[i]] = s[i];
  }

  return ans.join('');
}

console.log(restoreString('codeleet', [4, 5, 6, 7, 0, 2, 1, 3]));
console.log(restoreString('acb', [0, 2, 1]));
console.log(restoreString('abcdef', [0, 1,2,3,4,5]));

File: java/ShuffleString.java
class ShuffleString {
    public static void main(String[] args){
        ShuffleString tester = new ShuffleString();
        
        int[] arr = {4,5,6,7,0,2,1,3};
        System.out.println(tester.restoreString("codeleet", arr));
    }

    public String restoreString(String s, int[] indices) {

        char[] arr = new char[s.length()];

        for(int i = 0; i < indices.length ; i++) {
            arr[indices[i]] = s.charAt(i);
        }

        return String.valueOf(arr);
    }
}